### PR TITLE
[Azure Search] Update indexer and ServiceStats limits

### PR DESCRIPTION
**Pull request by enjaimes**

### CODE REVIEW

**Review by arv100kri — APPROVED** [review]

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/ServiceLimits.cs [hunk 1]
```diff
@@ -37,11 +37,15 @@ public ServiceLimits()
         /// <param name="maxComplexCollectionFieldsPerIndex">The maximum number
         /// of fields of type Collection(Edm.ComplexType) allowed in an
         /// index.</param>
-        public ServiceLimits(int? maxFieldsPerIndex = default(int?), int? maxFieldNestingDepthPerIndex = default(int?), int? maxComplexCollectionFieldsPerIndex = default(int?))
+        /// <param name="maxComplexObjectsInCollectionsPerDocument">The maximum
+        /// number of objects in complex collections allowed per
+        /// document.</param>
+        public ServiceLimits(int? maxFieldsPerIndex = default(int?), int? maxFieldNestingDepthPerIndex = default(int?), int? maxComplexCollectionFieldsPerIndex = default(int?), int? maxComplexObjectsInCollectionsPerDocument = default(int?))
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/ServiceLimits.cs]: This is a binary breaking change did you guys intend to break?

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/IndexerExecutionInfo.cs [hunk 2]
```diff
@@ -37,11 +37,13 @@ public IndexerExecutionInfo()
         /// in-progress indexer execution.</param>
         /// <param name="executionHistory">History of the recent indexer
         /// executions, sorted in reverse chronological order.</param>
-        public IndexerExecutionInfo(IndexerStatus status = default(IndexerStatus), IndexerExecutionResult lastResult = default(IndexerExecutionResult), IList<IndexerExecutionResult> executionHistory = default(IList<IndexerExecutionResult>))
+        /// <param name="limits">The execution limits for the indexer.</param>
+        public IndexerExecutionInfo(IndexerStatus status = default(IndexerStatus), IndexerExecutionResult lastResult = default(IndexerExecutionResult), IList<IndexerExecutionResult> executionHistory = default(IList<IndexerExecutionResult>), IndexerLimits limits = default(IndexerLimits))
```
Comment (weshaggard) [inline sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/IndexerExecutionInfo.cs]: This is a binary breaking change, assuming you have already shipped the older api. If you haven't shipped it then this is fine but if you have shipped it then you need to add an overload to ensure there isn't a breaking change.

**Review by arv100kri — CHANGES_REQUESTED** [review]
Because we deemed that this is a strictly binary breaking change from .NET code caller's perspective, let's not check this in yet, without revving a major version change

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/IndexerExecutionInfo.cs (hunk 2 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/IndexerExecutionInfo.cs]: @weshaggard - We've considered making this breaking change, which is why @enjaimes has bumped up the major version and make a release later in the month of August.
Please do merge this PR in whenever possible.

**Comment by arv100kri** [comment]
> Because we deemed that this is a strictly binary breaking change from .NET code caller's perspective, let's not check this in yet, without revving a major version change

Thanks @enjaimes for bumping up the major version!

**Review by arv100kri — APPROVED** [review]

File: sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/ServiceLimits.cs (hunk 1 above)
Comment (arv100kri) [reply sdk/search/Microsoft.Azure.Search.Service/src/Generated/Models/ServiceLimits.cs]: Please see the other comment - we've accepted this as a breaking change!

**Review by weshaggard — APPROVED** [review]

**Pull request merged**